partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/ExprPathKey.kt

```diff
@@ -19,6 +19,6 @@ internal class ExprPathKey(
         val rootEvaluated = root.eval(record).check<StructValue<PartiQLValue>>()
         val keyEvaluated = key.eval(record).check<StringValue>()
         val keyString = keyEvaluated.value ?: error("String value was null")
-        return rootEvaluated[keyString] ?: missingValue()
+        return rootEvaluated[keyString] ?: throw TypeCheckException()
```

Comment:
>  Tuple navigation in tuples with duplicate attributes When the tuple t has multiple attributes a, the tuple
path navigation t.a will return the first instance of a. Note that for tuples whose order is defined by schema, this is well-defined, for unordered tuples, it is implementation defined which attribute is returned in permissive mode or an error in type checking mode, which is described in Section 4.1

It seems like right now we don't have the notion of ordered vs unordered struct during runtime. Maybe just keep a todo note here?